### PR TITLE
Fix `pytest-rerunfailures` installation by using apt instead of pip

### PR DESCRIPTION
**Pull request by Crola1702**
See https://github.com/ros2/ci/pull/744

I used the same template as `install_python3.Dockerfile.em` for pytest-rerunfailures package

### CODE REVIEW

File: ros_buildfarm/templates/devel/devel_task.Dockerfile.em [hunk 1]
```diff
@@ -66,7 +66,12 @@ RUN python3 -u /tmp/wrapper_scripts/apt.py update-install-clean -q -y git python
 RUN python3 -u /tmp/wrapper_scripts/apt.py update-install-clean -q -y python3-pip
 @# colcon-core.package_identification.python needs at least setuptools 30.3.0
 @# pytest-rerunfailures enables usage of --retest-until-pass
-RUN pip3 install -U setuptools==59.6.0 pytest-rerunfailures
+@(TEMPLATE(
+    'snippet/install_pytest-rerunfailures.Dockerfile.em',
+    os_name=os_name,
+    os_code_name=os_code_name,
+))@
+RUN pip3 install -U setuptools==59.6.0
```
[the remark this answers is not in the document]
Comment (clalancette) [reply ros_buildfarm/templates/devel/devel_task.Dockerfile.em:74]: No, all of focal is gone, so we don't need to worry about it anymore.

File: ros_buildfarm/templates/snippet/install_pytest-rerunfailures.Dockerfile.em [hunk 2]
```diff
@@ -0,0 +1,4 @@
+@[if os_name == 'debian' or os_name == 'ubuntu' and os_code_name in ['jammy', 'noble']]@
```
[the remark this answers is not in the document]
Comment (clalancette) [reply ros_buildfarm/templates/snippet/install_pytest-rerunfailures.Dockerfile.em:1]: I honestly wouldn't even bother with the `os_code_name` check.  All of the Debian or Ubuntu distributions we support have this package in them.

**Review by clalancette — COMMENTED** [review]
This change looks reasonable to me.  Is there any way to test it without deploying it?

**Comment by Crola1702** [comment]
> Is there any way to test it without deploying it?

Maybe @kscottz or @nuclearsandwich know better

**Comment by nuclearsandwich** [comment]
> Is there any way to test it without deploying it?

You can run a devel job locally when set up:

```
mkdir /tmp/devel-test
generate_devel_script.py https://raw.githubusercontent.com/ros2/ros_buildfarm_config/ros2/index.yaml rolling default rcl ubuntu jammy amd64 > /tmp/devel-test/run.sh
```

Since the ros_buildfarm scripts assume that your local branch name when generating is available in the origin remote, you either need to change origin or push the branch to the root repository rather than a fork. I did the latter in this case.

**Review by nuclearsandwich — APPROVED** [review]
There's a small part of me which feels snippeting this is overkill, but not enough to block it as is. The local test I outlined gets the official Works On My Machine stamp of approval as well.

**Comment by cottsay** [comment]
This did regress CI after all. The job "ROS 1 Devel (3.6, colcon)" fails with:
```
  #22 0.326 Hit:1 http://repositories.ros.org/ubuntu/testing focal InRelease
  #22 0.916 Hit:2 http://security.ubuntu.com/ubuntu focal-security InRelease
  #22 0.956 Hit:3 http://archive.ubuntu.com/ubuntu focal InRelease
  #22 1.053 Hit:4 http://archive.ubuntu.com/ubuntu focal-updates InRelease
  #22 1.148 Hit:5 http://archive.ubuntu.com/ubuntu focal-backports InRelease
  #22 1.243 Hit:6 http://archive.ubuntu.com/ubuntu focal-security InRelease
  #22 1.300 Reading package lists...
  #22 2.146 Reading package lists...
  #22 2.960 Building dependency tree...
  #22 3.134 Reading state information...
  #22 3.149 Package python3-pytest-rerunfailures is not available, but is referred to by another package.
  #22 3.149 This may mean that the package is missing, has been obsoleted, or
  #22 3.149 is only available from another source
  #22 3.149 
  #22 3.152 E: Package 'python3-pytest-rerunfailures' has no installation candidate
```

**Pull request merged**